server/src/test/java/io/crate/planner/operators/EquiJoinDetectorTest.java

```diff
@@ -110,4 +110,11 @@ public void test_not_hash_join_possible_if_join_condition_refers_to_columns_from
         joinCondition = sqlExpressions.asSymbol("t1.x = t1.i");
         assertThat(EquiJoinDetector.isHashJoinPossible(JoinType.INNER, joinCondition), is(false));
     }
+
+    // tracks a bug : https://github.com/crate/crate/issues/15613
+    @Test
+    public void test_equality_expression_followed_by_case_expression() {
+        Symbol joinCondition = sqlExpressions.asSymbol("t1.a = t1.a AND CASE 1 WHEN t1.a THEN false ELSE t2.b in (t2.b) END");
+        assertThat(EquiJoinDetector.isHashJoinPossible(JoinType.INNER, joinCondition), is(false));
+    }
```

Comment:
`t1.a = t1.a AND CASE 1 WHEN t1.a THEN false ELSE t2.b in (t2.b) END` is not hash join possible since `t1.a = t1.a` is equivalent to `true`. Changing `t1.a = t1.a` => `t1.a = t2.b`, I expect it to be hash join possible but it will still return `false`, which is due to https://github.com/crate/crate/blob/7942990b1773b685dc5580b31ca73dcb6969ef4a/server/src/main/java/io/crate/planner/operators/EquiJoinDetector.java#L105-L107

This is out of scope here, will open a feature for this.

Reply:
Actually there is a test failure, I think I need to handle it.